provisioning/device/samples/solutions/BestPracticeSampleX509/readme.md

```diff
@@ -0,0 +1,139 @@
+# Best Practice Solution Sample with X.509 Authentication
+
+## Objective
+
+This document demonstrates a real-world example to help you get started with building your own custom IoT cloud solution with X.509 certificate-based authentication. See [here](https://learn.microsoft.com/azure/iot-dps/concepts-service#attestation-mechanism) to review available attestation mechanisms.
+
+## Example
+
+Contoso is a manufacturing company that wants to manage various types of products remotely.
+
+The business needs of Contoso are:
+
+- Provision devices securely at scale.
+- Group devices by production line.
+- Collect data from devices.
+- Manage devices remotely (e.g., controlling the temperature of a thermostat).
+
+## Recommended Solution
+
+![solution](media/auth_flow_diagram.png)
```

Comment:
auth flow and the boot sequence diagram seems to have different info. Let's double check the items in the comments for the boot sequence. For the Auth flow, is it targeted to identify how the scripts are creating the authentication certs and uploads that will be used in the boot sequence?

Maybe using #A, B. C etc for authentication setup, and #1,2,3 for boot sequence?